processor/filterprocessor/README.md

```diff
@@ -35,6 +35,11 @@ For logs:
   A match occurs if the record matches any expression in this given list.
 - `bodies`: Bodies defines a list of possible log bodies to match the logs against.
   A match occurs if the record matches any expression in this given list.
+- `min_severity`: MinSeverity defines the minimum severity with which a log record should match.
+  e.g. if this is "WARN", all log records with "WARN" severity and above (WARN[2-4], ERROR[2-4], FATAL[2-4]) are matched.
+  The list of valid severities that may be used for this option can be found [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#displaying-severity)
+  By default, logs with undefined severity are not matched.
+- `match_undefined_severity`: MatchUndefinedSeverity defines whether to match logs with undefined severity or not when using the `min_severity` matching option. If `min_severity` is not specified, this option does nothing. If `match_undefined_severity` is set to true, log records with no severity will be matched. If set to false, log records with no severity will not be matched.
```

Comment:
I'm not sure we need this option. The data model has an opinion on where "undefined" severity ranks vs other severities levels, so I think we should just support it according to that ordering.

Reply:
I was really on the fence about this one, but IMO it's really useful from a user perspective, because "undefined" severity can come up in exceptional circumstances, e.g. I set up a filelog recevier, with a severity parsing operator. But maybe I mess up and typo one of the severities or something.

Now I'm dropping logs, because they are now undefined severity and I'm filtering everything below "info" level. But maybe those logs are actually really important, just the severity can't be parsed.

I think it's a really nice to have this option when you need to be more fault-tolerant.

Reply:
I can appreciate the use case, but I don't think we should have a separate setting. What do you think about reworking the config a bit?
```yaml
log_severity:
  min: INFO
  match_undefined: true
```

Reply:
Reworked the config as suggested; I used `severity_number` instead of `log_severity` to avoid stuttering and potential confusion with severity_texts.